gradio/pipelines_utils.py

```diff
@@ -482,3 +482,321 @@ def is_diffusers_pipeline_type(pipeline, class_name: str):
             "postprocess": lambda r: r["images"][0],
         }
     raise ValueError(f"Unsupported diffusers pipeline type: {type(pipeline)}")
+
+
+def handle_transformers_js_pipeline(pipeline: Any) -> Dict[str, Any]:
+    try:
+        from transformers_js_py import as_url, read_audio  # type: ignore
+    except ImportError as ie:
+        raise ImportError(
+            "transformers_js_py not installed. Please add `transformers_js_py` to the requirements option of your Gradio-Lite app"
+        ) from ie
+
+    if pipeline.task == "audio-classification":
+        return {
+            "inputs": [
+                components.Audio(
+                    type="filepath",
+                    label="Input",
+                    render=False,
+                ),
+            ],
+            "outputs": components.Label(label="Class", render=False),
+            "preprocess": lambda i: (
+                read_audio(
+                    i, pipeline.processor.feature_extractor.config["sampling_rate"]
+                ),
+            ),
+            "postprocess": lambda r: {i["label"]: i["score"] for i in r},
+        }
+    if pipeline.task == "automatic-speech-recognition":
+        return {
+            "inputs": [
+                components.Audio(
+                    type="filepath",
+                    label="Input",
+                    render=False,
+                ),
+            ],
+            "outputs": components.Textbox(label="Output", render=False),
+            "preprocess": lambda i: (
+                read_audio(
+                    i, pipeline.processor.feature_extractor.config["sampling_rate"]
+                ),
+            ),
+            "postprocess": lambda r: r["text"],
+        }
+    # if pipeline.task == "depth-estimation":
```

Comment:
Now there is no Gradio component that directly takes and displays the depth map as a 3D model, while we can do it by writing a some amount code like this: https://huggingface.co/spaces/whitphx/gradio-lite-transformers-js-depth-estimation/blob/c7f86e976ddea4d573e98e5c42ceff524da49fe8/index.html#L35-L113 or this: https://huggingface.co/spaces/gradio/dpt-depth-estimation-3d-obj/blob/main/app.py

Is this a good time to consider to support depth map somehow, e.g. extending `gr.Model3D` or creating `gr.DepthMap` (or more generally `gr.PointCloud`)? @abidlabs

Reply:
Interesting question @whitphx. From a user interaction perspective, is a depth map just an image? If so, then I think we should just use the `gr.Image()` component for this (we may need to do some postprocessing to produce this image). However, if there's some interactions that depth maps provide, then we could definitely create a new component (I'm not totally sure). cc @dylanebert

Reply:
The depth map data is just a 1-channel NumPy array and it can be displayed as a gray-scale image.
Also we can construct 3D model (point cloud) from it.

I posted a demo video here: https://twitter.com/whitphx/status/1765346863069507876
The first output is the gray-scale image of the depth map with `gr.Image()`, and the second is a reconstructed point cloud with `gr.Model3D`.

So,
* To display the grayscale image, it's straightforward. Just pass the NumPy array to `gr.Image()`. No pre- or postprocess is needed.
* To display it as a 3D model that users can interact with, we need to convert the data from depth map (or point cloud) to a 3D model data that `gr.Model3D` can take, such as `*glb` file. The apps linked above does this by user-defined functions. My suggestion here is to support it natively.

Reply:
So like if you pass in an RGB image + grayscale depth map to gr.Model3D, it should convert it to a 3D object?

Reply:
Yes.
(and the RGB image can be optional)

Reply:
I believe this would be better suited for a custom component instead of baking it into the `gr.Model3D`. The reason being that we don't want to complicate the signature for any components' postprocess methods. Generally speaking, we try to limit the accepted formats for a `.postprocess()` to be a single type or variants of a single type, e.g. currently `gr.Model3D` accepts str filepaths only. In some cases, we allow variants such as `gr.Image` which can handle str filepaths, numpy arrays, or PIL images, but this is still pretty easy for users to grok. I think expanding the `gr.Mdoel3D` postprocess() to allow multiple images in this manner would make the the expected behavior of a component harder to understand. However, creating a custom component, e.g. `DepthMapper` to take in a tuple of images and render the 3D object should be quite straightforward.

Reply:
Makes sense.
I just wondered if it's worth supporting as a core component according to its demand, and if yes, it can be used in `from_pipeline()` for the depth-estimation task.
Then the answer would be no, so I will just use `gr.Image` in `from_pipeline` for now. (Apart from this, I may create a custom component as you advised btw.)